### PR TITLE
Small Changes to Address HTML & JS

### PR DESCRIPTION
**Pull request by MatthewKennedy**
### Frontend

- Updated address.js file to add `required` property to states input text field when it is used.
- Use `<abbr class="req">*</abbr>` for all elements in the address from that are used to indicate that a field is required. (Use `req` for the class so that it can be targeted easier in javascript and not hooking on to other elements with the `required` class).
- Added `form-control` class to elements that load hidden, so that if they get shown they are formatted the same as the existing elements on the page.
- Remove unused address erb files from pre address book.

### Backend

- Added into the countries admin config page the ability to toggle zip code required.

### CODE REVIEW

File: frontend/app/assets/javascripts/spree/frontend/checkout/address.js [hunk 1]
```diff
@@ -62,7 +58,7 @@ Spree.ready(function ($) {
           statesWithBlank = [{name: '', id: ''}].concat(states)
           $.each(statesWithBlank, function (idx, state) {
             var opt = $(document.createElement('option')).attr('value', state.id).html(state.name)
-            if (selected === state.id) {
+            if (selected == state.id) {
```
Comment (damianlegawiec) [inline frontend/app/assets/javascripts/spree/frontend/checkout/address.js]:
```suggestion
            if (parseInt(selected) === parseInt(state.id)) {
```

will do the trick

**Review by damianlegawiec — COMMENTED** [review]
Thank you @MatthewKennedy 👍 awesome work!

**Comment by MatthewKennedy** [comment]
There is still a bug in Spree 4 where this fix works for U.S, but not for for other countries, I think its taking the default country.

I'm not sure if this is related to the issue, but looking at the code in the javascript tags there are listed all states for the U.S, even though the country has been changed to Canada and the HTML is listing the Canadian states.
 
Screenshot 2019-09-17 at 13 28 35

**Comment by damianlegawiec** [comment]
@MatthewKennedy could you rebase your PR against master? Thanks!

**Comment by MatthewKennedy** [comment]
@damianlegawiec I noticed the old `address/_form.html.erb` file and folder are still in the views although they are not used any longer, do you want me to remove them also in this PR?

**Comment by damianlegawiec** [comment]
@MatthewKennedy I think this is still in use in the Admin Panel, if not please create another PR, thank you!

**Comment by MatthewKennedy** [comment]
@damianlegawiec As far as I can tell the backend has its own address view file in shared folder.

**Pull request closed without merging**